File: taller/JCZhang-Taller/JCZhang-Taller-Java/src/main/java/models/Trabajador.java
package models;

public class Trabajador extends Persona {

    String nombre;
    int aniosExp;
    int horasDiarias;

    public Trabajador(String nombre, int aniosExp, int horasDiarias) {
        this.nombre = nombre;
        this.aniosExp = aniosExp;
        this.horasDiarias = horasDiarias;
    }


    @Override
    void saludar() {
        System.out.println("Trabajador: Hola! Me llamo "+ nombre+ " y soy un humilde trabajador");
    }

    public void trabajar(){
        System.out.println("Trabajador: Hola soy $nombre y estoy trabajando");
    }

    public void descansar(){
        System.out.println("Trabajador: Hola soy $nombre estoy descansando");
    }

    public void comer(){}

    @Override
    public String toString() {
        return "Trabajador (" +nombre+" con " + aniosExp +"años de experiencia trabaja " + horasDiarias + "horas al dia)";

    }
}

File: taller/JCZhang-Taller/JCZhang-Taller-Kotlin/src/main/kotlin/models/Trabajador.kt
package models

open class Trabajador(nombre: String, aniosExp: Int, var horasDiarias: Int ) : Persona(id, nombre, aniosExp) {

    fun Trabajador(nombre: String, aniosExp: Int, horasDiaras: Int) {
        this.nombre = nombre
        this.aniosExp = aniosExp
        this.horasDiarias = horasDiaras
    }

    override fun saludar() {
        println("Trabajador: Hola! Mi nombre es $nombre y soy un humilde trabajador ")
    }
    fun trabajar(){
        println("Trabajador: Hola soy $nombre y estoy trabajando")
    }
    fun descansar(){
        println("Trabajador: Hola soy $nombre estoy descansando")
    }
    open fun comer(){}
    override fun toString(): String {
        return "Trabajador( $nombre con $aniosExp años de experiencia trabaja $horasDiarias horas al dia)"
    }


}